Code/GraphMol/ChemReactions/ReactionRunner.cpp

```diff
@@ -854,4 +882,192 @@ std::vector<MOL_SPTR_VECT> run_Reactants(const ChemicalReaction &rxn,
   return productMols;
 }  // end of ChemicalReaction::runReactants()
 
+// Generate a product based on a SINGLE reactant
+std::vector<MOL_SPTR_VECT> run_Reactant(const ChemicalReaction &rxn,
+                                        const ROMOL_SPTR &reactant,
+                                        unsigned int reactantIdx) {
```

Comment:
That is certainly possible outside of the function, however it is still necessary to supply a reactantIdx as the reactant may actually match multiple reactant templates.

The intent is that this is really a utility function to help analyze reactions.  I.e. run a bunch of reagents through to see what side-chains they add.